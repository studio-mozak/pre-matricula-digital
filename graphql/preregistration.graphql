extend type Query @guard(with: ["prematricula"]) {

  """
  Lista as pré-matrículas.
  """
  preregistrations(
    """
    Retorna apenas as pré-matrículas onde o **protocolo** ou o **nome do(a) aluno(a)** contenha `search`.
    """
    search: String @slug @search(columns: ["protocol", "student.slug"])
    process: ID @eq(key: "process_id")
    processes: [ID!] @in(key: "process_id")
    school: ID @eq(key: "school_id")
    schools: [ID!] @in(key: "school_id")
    grade: ID @eq(key: "grade_id")
    grades: [ID] @in(key: "grade_id")
    period: ID @eq(key: "period_id")
    type: PreRegistrationType @eq(key: "preregistration_type_id")
    status: PreRegistrationStatus @eq(key: "status")
    sort: PreRegistrationSort @scope(name: "sort")
    year: Int @search(columns: ["process.school_year_id"])
  ): [PreRegistration!]! @paginate

}

extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Retorna a pré-matrícula do protocolo.
  """
  preregistrationByProtocol(
    protocol: String @eq
    withVacancy: Boolean
  ): PreRegistration @first

  """
  Retorna a inscrição usando o código de verificação.
  """
  preregistrationByCode(
    code: String @eq
  ): PreRegistration @first

  getStudentMatches(
    stage: ID!
    name: String @slug
    dateOfBirth: Date @rename(attribute: "date_of_birth")
    cpf: String
    rg: String
    birthCertificate: String @rename(attribute: "birth_certificate")
  ): [StudentMatch!]!

  summonNextInLine(
    process: ID! @eq(key: "process_id")
    school: ID! @eq(key: "school_id")
    grade: ID! @eq(key: "grade_id")
    period: ID! @eq(key: "period_id")
  ): PreRegistration @first(scopes: ["nextInLine", "waiting"])

  matchByStudentCpf(
    cpf: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentRg(
    rg: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentBirthCertificate(
    birthCertificate: String! @eq(key: "birth_certificate")
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentNameAndDateOfBirth(
    name: String! @slug @eq(key: "slug")
    dateOfBirth: Date! @eq(key: "date_of_birth")
    year: Int! @eq
  ): RegistrationMatch @first

  findPreRegistrationsByStudentCpf(
    cpf: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentRg(
    rg: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentBirthCertificate(
    birthCertificate: String! @rename(attribute: "birth_certificate")
  ): [PreRegistration!]!

  findPreRegistrationsByStudentNameAndDateOfBirth(
    name: String! @slug
    dateOfBirth: Date! @rename(attribute: "date_of_birth")
  ): [PreRegistration!]!

  findPreRegistrationsByInfo(
    cpf: String @rename(attribute: "responsible_cpf")
    email: String @rename(attribute: "responsible_email")
    protocol: String
  ): [PreRegistration!]!

  getPreRegistrationLinkedByEmail(
    email: String! @eq
  ): [PreRegistrationLinkedByEmail!]! @all

  searchLegacyStudentsByName(
    name: String!
    first: Int
  ): [LegacyStudent!]!
}

enum PreRegistrationType {
  """
  Rematrícula
  """
  REGISTRATION_RENEWAL @enum(value: 1)
  """
  Matrícula
  """
  REGISTRATION @enum(value: 2)
  """
  Lista de espera
  """
  WAITING_LIST @enum(value: 3)
}

enum PreRegistrationStatus {
  """
  Pré-matrícula aguardando (in)deferimento.
  """
  WAITING @enum(value: 1)

  """
  Pré-matrícula deferida.
  """
  ACCEPTED @enum(value: 2)

  """
  Pré-matrícula indeferida.
  """
  REJECTED @enum(value: 3)

  """
  Pré-matrícula convocada.
  """
  SUMMONED @enum(value: 4)

  """
  Pré-matrícula em confirmação.
  """
  IN_CONFIRMATION @enum(value: 5)
}

enum RelationType {
  """
  Mãe
  """
  MOTHER @enum(value: 1)

  """
  Pai
  """
  FATHER @enum(value: 2)

  """
  Responsável legal
  """
  GUARDIAN @enum(value: 3)

  """
  A própria pessoa
  """
  SELF @enum(value: 4)
}

enum PreRegistrationSort {
  DATE @enum(value: 1)
  POSITION @enum(value: 2)
  SCHOOL @enum(value: 3)
  NAME @enum(value: 4)
  DATE_OF_BIRTH @enum(value: 5)
}

type RegistrationMatch {
  id: ID!
  year: Int!
  school: ID!
  grade: ID!
  period: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
}

enum StudentMatchType {
  PRE_REGISTRATION_ALREADY_EXISTS @enum(value: 1)
  REGISTRATION_CURRENT_YEAR @enum(value: 2)
  REGISTRATION_LAST_YEAR @enum(value: 3)
  STUDENT_ALREADY_EXISTS @enum(value: 4)
}

type StudentMatch {
  id: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
  type: ProcessStageRestriction!
  process: Process
  registration: StudentMatchRegistration
}

type StudentMatchRegistration {
  year: Int!
  school: School!
  grade: Grade!
  period: Period
}

type PreRegistration {
  id: ID!
  type: PreRegistrationType! @rename(attribute: "preregistration_type_id")
  position: Int!
  parent: PreRegistration @belongsTo
  waiting: PreRegistration @belongsTo
  others: [PreRegistration]
  protocol: String!
  code: String!
  date: DateTime! @rename(attribute: "created_at")
  process: Process @belongsTo
  period: Period @belongsTo
  school: School @belongsTo
  grade: Grade @belongsTo
  schoolYear: SchoolYear @belongsTo
  classroom: Classroom @belongsTo
  inClassroom: Classroom @belongsTo
  student: Person @belongsTo
  responsible: Person @belongsTo
  relationType: RelationType! @rename(attribute: "relation_type_id")
  status: PreRegistrationStatus!
  observation: String
  fields: [PreRegistrationField!]! @hasMany
  stage: ProcessStage @belongsTo
  external: ExternalPerson @hasOne
  linked: [PreRegistrationLinkedByEmail!]! @hasMany
}

type PreRegistrationField {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  field: Field! @belongsTo
  value: String
}

type PreRegistrationLinkedByEmail {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  email: String!
}

extend type Mutation @guard(with: ["web", "sanctum", "prematricula"]) {

  newPreRegistration(
    input: PreRegistrationInput! @spread
  ): [PreRegistration!]!

  sendProtocolsByEmail(
    preregistrations: [ID!]!
    email: String!
  ): Boolean!

  keepOnTheWaitingList(
    id: ID!
    grade: ID
  ): Boolean!

  updateStudentInExternalSystem(
    preregistration: ID!
    cpf: Boolean
    rg: Boolean
    birthCertificate: Boolean
    name: Boolean
    dateOfBirth: Boolean
    gender: Boolean
    phone: Boolean
    mobile: Boolean
    address: Boolean
  ): Boolean!

  savePreRegistrationLinkedByEmail(
    preregistration: ID! @rename(attribute: "preregistration_id")
    email: String!
  ): PreRegistrationLinkedByEmail!

  deletePreRegistrationLinkedByEmail(
    id: ID! @whereKey
  ): PreRegistrationLinkedByEmail! @delete

}

extend type Mutation @guard {
  acceptPreRegistrations(
    ids: [ID!]!
    classroom: ID!
  ): [PreRegistration!]!

  rejectPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  summonPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  returnToWaitPreRegistrations(
    ids: [ID!]!
  ): [PreRegistration!]!

  updateStudent(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updateResponsible(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updatePreRegistration(
    protocol: String!
    grade: ID!
    school: ID!
    period: ID!
  ): PreRegistration

  updateAddress(
    protocol: String!
    address: AddressInput
  ): PreRegistration
}

input PreRegistrationFieldInput {
  field: ID!
  value: String
}

input AddressInput {
  postalCode: String! @rename(attribute: "postal_code")
  address: String!
  number: String!
  complement: String
  neighborhood: String!
  city: String
  stateAbbreviation: String
  cityIbgeCode: Int
  lat: Float! @rename(attribute: "latitude")
  lng: Float! @rename(attribute: "longitude")
  manualChangeLocation: Boolean! @rename(attribute: "manual_change_location")
}

input PreRegistrationWaitingListInput {
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
}

input PreRegistrationInput {
  type: PreRegistrationType!  @rename(attribute: "preregistration_type_id")
  process: ID! @rename(attribute: "process_id")
  stage: ID! @rename(attribute: "process_stage_id")
  grade: ID! @rename(attribute: "grade_id")
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
  optionalSchool: ID
  optionalPeriod: ID
  address: AddressInput!
  optionalAddress: AddressInput
  relationType: RelationType! @rename(attribute: "relation_type_id")
  responsible: [PreRegistrationFieldInput!]!
  student: [PreRegistrationFieldInput!]!
  externalPerson: ID @rename(attribute: "external_person_id")
  waitingList: [PreRegistrationWaitingListInput!]
}

type LegacyStudent {
  id: ID!
  name: String!
  cpf: String
  dateOfBirth: String @rename(attribute: "date_of_birth")
  responsibleName: String @rename(attribute: "responsible_name")
  responsibleCpf: String @rename(attribute: "responsible_cpf")
  responsibleDateOfBirth: String @rename(attribute: "responsible_date_of_birth")
  responsiblePhone: String @rename(attribute: "responsible_phone")
}
