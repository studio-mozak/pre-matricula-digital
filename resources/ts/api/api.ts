import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
