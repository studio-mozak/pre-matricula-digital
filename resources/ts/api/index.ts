export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
