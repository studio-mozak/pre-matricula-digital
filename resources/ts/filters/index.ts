import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
