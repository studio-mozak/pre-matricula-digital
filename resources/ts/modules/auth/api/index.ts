/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
