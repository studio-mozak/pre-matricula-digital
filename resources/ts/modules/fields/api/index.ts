/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
