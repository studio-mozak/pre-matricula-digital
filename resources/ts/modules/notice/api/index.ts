/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
