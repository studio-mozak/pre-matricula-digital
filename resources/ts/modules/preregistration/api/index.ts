/**
 * Exporting the Axios Instance
 */
export { http, graphql, rest } from '@/api';

/**
 * Exporting all modules
 */
export * as Preregistration from './services/graphql/preregistration';
export * as PreregistrationRest from './services/rest/preregistration';
