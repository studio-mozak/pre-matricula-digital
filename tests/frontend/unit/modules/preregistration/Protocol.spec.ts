import {
  PreRegistrationOverview,
  PreRegistrationResponsibleField,
  PreRegistrationStageProcess,
  PreRegistrationStudentField,
} from '@/modules/preregistration/types';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { Preregistration as PreregistrationApi } from '@/modules/preregistration/api';
import Protocol from '@/modules/preregistration/components/Protocol.vue';
import VTooltip from 'v-tooltip';
import { createTestingPinia } from '@pinia/testing';
import filters from '@/filters';
import { expect, vi } from 'vitest';

vi.mock('@/modules/preregistration/api');

interface ProtocolProps {
  preregistrations: PreRegistrationOverview[];
  process: PreRegistrationStageProcess;
  student: PreRegistrationStudentField;
  responsible: PreRegistrationResponsibleField;
}

const propsMock: ProtocolProps = {
  preregistrations: [],
  responsible: {
    relationType: null,
    address: {
      postalCode: '',
      address: '',
      number: '',
      complement: '',
      neighborhood: '',
      lat: 0,
      lng: 0,
      city: '',
      cityIbgeCode: 0,
      stateAbbreviation: '',
    },
    useSecondAddress: false,
    secondAddress: {
      postalCode: '',
      address: '',
      number: '',
      complement: '',
      neighborhood: '',
      lat: 0,
      lng: 0,
      city: '',
      cityIbgeCode: 0,
      stateAbbreviation: '',
    },
    responsible_name: null,
  } as PreRegistrationResponsibleField,
  student: {
    grade: null,
    period: null,
    school: null,
    useSecondSchool: false,
    secondSchool: null,
    secondPeriod: null,
    match: null,
    student_name: null,
    student_date_of_birth: '2015-01-01T00:00:00.000Z',
    student_cpf: 'string',
    student_rg: 'string',
    student_marital_status: 1,
    student_place_of_birth: 1,
    student_birth_certificate: 'string',
    student_gender: 1,
    student_email: 'string',
    student_phone: 'string',
    student_mobile: 'string',
  },
  process: {
    forceSuggestedGrade: true,
    gradeAgeRangeLink: null,
    id: '1',
    messageFooter:
      'Acompanhe a publicação do resultado classificatório e convocação no site da Prefeitura.',
    name: 'PROCESSO DE TESTES 2022',
    showPriorityProtocol: true,
    grades: [
      {
        endBirth: '2016-03-31',
        id: '22',
        name: '1º Ano',
        startBirth: '2015-03-31',
      },
    ],
    periods: [
      {
        id: '1',
        name: 'Matutino',
      },
    ],
    schoolYear: { year: 2022 },
    schools: [
      {
        id: '12345',
        latitude: 0,
        longitude: 0,
        name: 'ESCOLA MUNICIPAL DE ENSINO FUNDAMENTAL',
      },
    ],
    vacancies: [
      {
        available: 8,
        grade: 6,
        period: 1,
        school: 12345,
        total: 25,
      },
    ],
    fields: [
      {
        field: {
          group: 'RESPONSIBLE',
          id: '1',
          internal: 'responsible_name',
          name: 'Nome do(a) responsável',
          options: [],
          type: 'TEXT',
        },
        id: '262',
        order: '1',
        required: true,
        weight: 0,
      },
      {
        field: {
          group: 'STUDENT',
          id: '9',
          internal: 'student_name',
          name: 'Nome',
          options: [],
          type: 'TEXT',
        },
        id: '267',
        order: '2',
        required: true,
        weight: 0,
      },
    ],
  },
};

const onMounted = async (props: ProtocolProps) => {
  const wrapper = mount(Protocol, {
    props,
    global: {
      plugins: [createTestingPinia()],
      stubs: ['x-field'],
      directives: {
        tooltip: VTooltip,
      },
      provide: {
        $filters: filters,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('Protocol', () => {
  let wrapper: VueWrapper<InstanceType<typeof Protocol>>;

  beforeEach(async () => {
    wrapper = await onMounted(propsMock);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('should call window.print', () => {
    const spy = (window.print = vi.fn());

    wrapper.vm.print();

    expect(spy).toBeCalled();
    expect(spy).toBeCalledTimes(1);
  });

  test('openModalEmail function must set the correct variables', () => {
    wrapper.vm.openModalEmail();

    expect(wrapper.vm.modal.email).toBe(true);
    expect(wrapper.vm.email).toBe(propsMock.responsible.responsible_email);
  });

  describe('sendEmail', () => {
    test('should call sendEmail successfully', async () => {
      wrapper.vm.modal.email = true;

      vi.mocked(PreregistrationApi.postSendEmail).mockResolvedValueOnce(true);

      wrapper.vm.sendEmail('test@email.com');

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(wrapper.vm.modal.email).toBe(false);
    });

    test('should call sendEmail unsuccessfully', async () => {
      wrapper.vm.modal.email = true;

      vi.mocked(PreregistrationApi.postSendEmail).mockResolvedValueOnce(false);

      wrapper.vm.sendEmail('test@email.com');

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(wrapper.vm.modal.email).toBe(true);
    });
  });

  test('when I call submit function, should emit finish event', () => {
    wrapper.vm.submit();

    expect(wrapper.emitted().finish).toBeTruthy();
  });
});
