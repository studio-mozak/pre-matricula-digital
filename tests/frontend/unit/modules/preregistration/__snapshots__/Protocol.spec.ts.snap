// Vitest Snapshot v1

exports[`Protocol > should match snapshot 1`] = `
<div
  class="container-pmd"
  data-v-ca987901=""
>
  <h2
    class="font-muli-20-primary mb-4 text-center"
    data-v-ca987901=""
  >
     Sua inscrição foi realizada com sucesso! 
  </h2>
  <p
    class="mb-4"
    data-v-ca987901=""
  >
     Para acompanhar a atualização da Pré-matrícula, não esqueça de anotar/salvar o(s) código(s) do(s) 
    <strong
      data-v-ca987901=""
    >
      protocolo(s)
    </strong>
     gerado(s). 
  </p>
  <div
    class="m-auto"
    data-v-ca987901=""
    style="max-width: 340px;"
  >
    <!--v-if-->
    <!--v-if-->
    
    
  </div>
  <div
    class="mt-5 text-center"
    data-v-ca987901=""
  >
    Acompanhe a publicação do resultado classificatório e convocação no site da Prefeitura.
  </div>
  <div
    class="text-center mt-5 mb-4"
    data-v-ca987901=""
  >
    <div
      class="row mb-5"
      data-v-ca987901=""
    >
      <div
        class="col-12 col-md-4 offset-md-2"
        data-v-ca987901=""
      >
        <button
          class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-primary border-primary x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 mb-2"
          data-v-ca987901=""
          role="button"
          style="font-size: 20px;"
          tabindex="0"
          type="button"
        >
          <span
            class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
          >
            <span
              class="block"
            >
              Imprimir
            </span>
          </span>
          <transition-stub
            appear="false"
            css="true"
            name="x-transition--fade"
            persisted="false"
          >
            <!---->
          </transition-stub>
        </button>
      </div>
      <div
        class="col-12 col-md-4"
        data-v-ca987901=""
      >
        <button
          class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-primary border-primary x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
          data-v-ca987901=""
          role="button"
          style="font-size: 20px;"
          tabindex="0"
          type="button"
        >
          <span
            class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
          >
            <span
              class="block"
            >
              Enviar por e-mail
            </span>
          </span>
          <transition-stub
            appear="false"
            css="true"
            name="x-transition--fade"
            persisted="false"
          >
            <!---->
          </transition-stub>
        </button>
      </div>
    </div>
    <div
      class="row"
      data-v-ca987901=""
    >
      <div
        class="col-12 col-md-5 offset-md-1 mb-4"
        data-v-ca987901=""
      >
        <button
          class="x-btn x-btn-item non-selectable no-outline x-btn--outline x-btn--rectangle text-primary border-primary x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
          data-v-ca987901=""
          role="button"
          style="font-size: 20px;"
          tabindex="0"
          type="button"
        >
          <span
            class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
          >
            <span
              class="block"
            >
              Nova Inscrição
            </span>
          </span>
          <transition-stub
            appear="false"
            css="true"
            name="x-transition--fade"
            persisted="false"
          >
            <!---->
          </transition-stub>
        </button>
      </div>
      <div
        class="col-12 col-md-5 mb-4"
        data-v-ca987901=""
      >
        <button
          class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100 border-primary"
          data-v-ca987901=""
          role="button"
          style="font-size: 20px;"
          tabindex="0"
          type="button"
        >
          <span
            class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
          >
            <span
              class="block"
            >
              Voltar para a tela inicial
            </span>
          </span>
          <transition-stub
            appear="false"
            css="true"
            name="x-transition--fade"
            persisted="false"
          >
            <!---->
          </transition-stub>
        </button>
      </div>
    </div>
  </div>
  <!--v-if-->
</div>
`;
